src/hooks/hipchat.js

```diff
@@ -0,0 +1,52 @@
+const Hook = require("../hook");
+
+const MXControlEvent = require("../events/mxcontrol");
+const MXControlHandler = require("../handlers/mxcontrol");
+const minimist = require("minimist");
+
+//a primer to this type of function
+//(cause this shit is difficult the first time)
```

Comment:
is it really that confusing? the point is just to both be able to use parameter destructuring and also have access to the original object

Reply:
Excruciatingly. The syntax doesn't --feel-- very explicit. It --feels-- too loose. These are just --feels,-- however.

The second time around isn't so bad